src/Propel/Generator/Behavior/Archivable/ArchivableBehavior.php

```diff
@@ -128,6 +131,11 @@ protected function addArchiveTable(): void
             // copy the indices
             foreach ($table->getIndices() as $index) {
                 $copiedIndex = clone $index;
+                // Some platforms require unique index names across whole database
+                // by unsetting the name, Propel will generate a unique name based on table and columns
+                if ($platform instanceof PgsqlPlatform || $platform instanceof SqlitePlatform) {
```

Comment:
Can we move this check to separate method? At least it will be more clan in terms of code.